app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -145,117 +233,137 @@ export const snapKeyringBuilder = (
-                await showSuccess(
-                  controllerMessenger,
-                  snapId,
-                  {
-                    icon: IconName.UserCircleAdd,
-                    title: t('snapAccountCreated'),
-                  },
-                  {
-                    message: t('snapAccountCreatedDescription') as string,
-                    address,
-                    learnMoreLink,
-                  },
-                );
-              }
-
-              // User has clicked on "OK"
-              trackSnapAccountEvent(
-                MetaMetricsEventName.AddSnapAccountSuccessClicked,
-              );
-            } catch (e) {
-              const error = (e as Error).message;
-
-              await showError(
+              await showSuccess(
                 controllerMessenger,
                 snapId,
                 {
                   icon: IconName.UserCircleAdd,
-                  title: t('snapAccountCreationFailed'),
+                  title: t('snapAccountCreated'),
                 },
                 {
-                  message: t(
-                    'snapAccountCreationFailedDescription',
-                    snapName,
-                  ) as string,
+                  message: t('snapAccountCreatedDescription') as string,
+                  address,
                   learnMoreLink,
-                  error,
                 },
               );
-
-              trackSnapAccountEvent(MetaMetricsEventName.AccountAddFailed);
-
-              throw new Error(
-                `Error occurred while creating snap account: ${error}`,
+              // User has clicked on "OK"
+              trackSnapAccountEvent(
+                MetaMetricsEventName.AddSnapAccountSuccessClicked,
               );
             }
-          } else {
-            // User has cancelled account creation
-            await handleUserInput(confirmationResult);
 
-            throw new Error('User denied account creation');
-          }
-        } finally {
-          // We do not have a `else` clause here, as it's used if the request was
-          // canceled by the user, thus it's not a "fail" (not an error).
-          if (confirmationResult) {
             trackSnapAccountEvent(MetaMetricsEventName.AccountAdded);
+          } catch (e) {
+            // Error occurred while naming the account
+            const error = (e as Error).message;
+
+            await showError(
+              controllerMessenger,
+              snapId,
+              {
+                icon: IconName.UserCircleAdd,
+                title: t('snapAccountNamingFailed'),
+              },
+              {
+                message: t(
+                  'snapAccountNamingFailedDescription',
+                  snapName,
+                ) as string,
+                learnMoreLink,
+                error,
+              },
+            );
+
+            trackSnapAccountEvent(MetaMetricsEventName.AccountNameFailed);
```

Comment:
Side note: Since this is a new event, we would have to create/add it to the Segment schemas too.

Reply:
Was this event validated by Alex?

Reply:
Let's remove this event, events must be defined by product

Reply:
Okay! I didn't know the events needed to be defined by product. I just added it because it did not make sense to use the add account failed event here, but we can safely not use any event.
[3bb7c4c](https://github.com/MetaMask/metamask-extension/pull/25191/commits/3bb7c4ca0e865e243b62679f6c1f88dbf7885631)